cfme/tests/cloud_infra_common/test_tag_objects.py

```diff
@@ -1,128 +1,156 @@
-    'Infra VMs': Vm,
-    'Infra Templates': Template,
-    'Infra Hosts': HostCollection,
-    'Infra Clusters': ClusterCollection,
-    'Infra Stores': DatastoreCollection,
-
-    'Cloud Providers': CloudProvider,
-    'Cloud Instances': Instance,
-    # 'Cloud Flavors': Flavor, # Test needs to be refactored along with tag mixin for widgets
-    # 'Cloud Availabity Zones': AvailabilityZone,  # Add back when all classes support widgets
-    'Cloud Tenants': TenantCollection
-}
-
-pytestmark = [
-    pytest.mark.parametrize("location", param_classes),
-    pytest.mark.tier(3)
-]
-
-
-def _navigate_and_check(location, appliance):
-    if issubclass(param_classes[location], BaseCollection):
-        cls_or_col = param_classes[location](appliance)
-    else:
-        cls_or_col = param_classes[location]
-    view = navigate_to(cls_or_col, 'All')
-    view.toolbar.view_selector.select('Grid View')
-    try:
-        entity = view.entities.get_first_entity()
-        entity.check()
-        return entity
-    except ItemNotFound:
-        pytest.skip("No Quadicon present, cannot test.")
 
-# TODO Replace navigation and item selection with widgets when all tested classes have them
 
+infra_test_items = [
+    ('infra_provider', InfraProvider),
+    ('vms', Vm),
+    ('templates', Template),
+    ('hosts', None),
+    ('clusters', None),
+    ('datastores', None)
+]
 
-def _tag_item_through_selecting(request, location, tag, appliance):
-    """Add a tag to an item with going through the details page.
+cloud_test_items = [
+    ('cloud_provider', CloudProvider),
+    ('instances', Instance),
+    ('flavors', Flavor),
+    ('availability_zones', AvailabilityZone),
+    ('cloud_tenants', None),
+    ('keypairs', None),
+    ('templates', Image)
+]
 
-    Prerequisities:
-        * Have a tag category and tag created.
-        * Be on the page you want to test.
 
-    Steps:
-        * Select any quadicon.
-        * Select ``Policy/Edit Tags`` and assign the tag to it.
-        * Click on the quadicon and verify the tag is assigned. (TODO)
-        * Go back to the quadicon view and select ``Policy/Edit Tags`` and remove the tag.
-        * Click on the quadicon and verify the tag is not present. (TODO)
-    """
-    _navigate_and_check(location, appliance)
+@pytest.fixture(scope='module')
+def infra_provider(request):
+    prov_filter = ProviderFilter(classes=[InfraProvider])
+    return setup_one_or_skip(request, filters=[prov_filter])
 
-    def _delete():
-        # Ignoring the result of the check here
-        _navigate_and_check(location, appliance)
-        mixins.remove_tag(tag)
-    request.addfinalizer(lambda: diaper(_delete))
-    mixins.add_tag(tag)
-    _delete()
 
+@pytest.fixture(scope='module')
+def cloud_provider(request):
+    prov_filter = ProviderFilter(classes=[CloudProvider],
+                                 required_fields=[['provisioning', 'stacks']])
+    return setup_one_or_skip(request, filters=[prov_filter])
```

Comment:
**REQUIRED**
Here as well

Reply:
I've restored cloud/infra provider fixtures, I think it is better to use them here, as it selects any provider, not just OpenStack or Vmware as I set before, second point if use pytest.mark.provider it adds provider info to name tests case, I would like to omit that(will be easier to track in polarion), one more thing in my case while running test cases to many time is taken by provider add/delete(for every second test case), test run lasts 1h longer.  

I would like to leave local cloud/infra provider fixtures if such usage is not deprecated, however, if there is another better way to do that I'll be happy to add updates
@quarckster  what do think?

Reply:
I'm ok with that, but I also would like to know what @mshriver, @jkrocil  think about it.

Reply:
Yes, this is fine. It could only be an issue if one of the fixtures was trying to set itself up while the other was still being used (since it would remove the other provider), but that is not the case here.